src/components/Calculator.js

```diff
@@ -1,44 +1,100 @@
-          <div className="1">1</div>
-          <div className="two">2</div>
-          <div className="three">3</div>
-          <div className="zero">0</div>
-          <div className="dot">.</div>
+          <button type="button" onClick={calculation} className="ac">
+            AC
+          </button>
+          <button type="button" onClick={calculation} className="plusOrMinus">
+            +/-
+          </button>
+          <button type="button" onClick={calculation} className="modelus">
+            %
+          </button>
+          <button type="button" onClick={calculation} className="seven">
+            7
+          </button>
+          <button type="button" onClick={calculation} className="eight">
+            8
+          </button>
+          <button type="button" onClick={calculation} className="nine">
+            9
+          </button>
+          <button type="button" onClick={calculation} className="four">
+            4
+          </button>
+          <button type="button" onClick={calculation} className="five">
+            5
+          </button>
+          <button type="button" onClick={calculation} className="six">
+            6
+          </button>
+          <button type="button" onClick={calculation} className="1">
+            1
+          </button>
+          <button type="button" onClick={calculation} className="two">
+            2
+          </button>
+          <button type="button" onClick={calculation} className="three">
+            3
+          </button>
+          <button type="button" onClick={calculation} className="zero">
+            0
+          </button>
+          <button type="button" onClick={calculation} className="dot">
+            .
+          </button>
         </div>
         <div className="right-section">
-          <div className="plus">/</div>
-          <div className="plus">x</div>
-          <div className="plus">-</div>
-          <div className="plus">+</div>
-          <div className="plus">=</div>
+          <button type="button" onClick={calculation} className="plus">
+            /
+          </button>
+          <button type="button" onClick={calculation} className="plus">
+            x
+          </button>
+          <button type="button" onClick={calculation} className="plus">
+            -
+          </button>
+          <button type="button" onClick={calculation} className="plus">
+            +
+          </button>
+          <button type="button" onClick={calculation} className="plus">
+            =
+          </button>
         </div>
       </div>
     </div>
-  </>
-);
+  );
 
-const Operand = () => (
-  <>
-    <div className="operand-input">
-      <input type="text" name="operand" value="0" />
-    </div>
-  </>
-);
+  function Operand(props) {
+    const state = props;
+    return (
+      <div className="operand-input">
+        <input
+          type="text"
+          name="operand"
+          value={state.data.next || state.data.total || 0}
+        />
```

Comment:
- [ ] [OPTIONAL] The negative sign is displayed on the wrong side for negative values. I would suggest you to rearrange the `state.data` values. Example snippet:

```js

 <div className="panel">
          {total}
          {operation}
          {next}
        </div>
```

Reply:
Thanks again. I will fix this issues